server/controllers/tripController.js

```diff
@@ -0,0 +1,152 @@
+   * @param {string} returnDate The response.
+   * @param {integer} cityNumber The response.
+   * @returns {object} The status and some data of the trip.
+   */
+  static async setTripType(trip, returnDate) {
+    if (returnDate) {
+      trip.returnDate = returnDate;
+      trip.tripType = 'Return';
+    } else {
+      trip.tripType = 'One-way';
+    }
+    return trip;
+  }
+
+  /**
+   * This method handle createTrip request.
+   * @param {object} req The user's request.
+   * @param {object} res The response.
+   * @returns {object} The status and some data of the trip.
+   */
+  static async createTrip(req, res) {
+    try {
+      const { body } = req;
+      const { lineManager } = req.user;
+      if (lineManager !== null) {
+        const myuserId = req.user.id;
+        const status1 = 'pending';
+        let newTrip = {
+          departure: body.departure,
+          destination: body.destination,
+          date: body.date,
+          reasons: body.reasons,
+          userId: myuserId,
+          status: status1
+        };
+        const { returnDate, destination } = body;
+        const cityNumber = destination.length;
+
+        newTrip = await TripController.setTripType(newTrip, returnDate, cityNumber);
+        const foundTrip = await tripHelpers.reasonsDate({
+          userId: myuserId,
+          reasons: body.reasons,
+          destination: body.destination,
+          date: body.date
+        });
+        if (foundTrip) {
+          return res.status(409).json({
+            status: 409,
+            data: 'You create this trip'
+          });
+        }
+        const saveTrip = await tripHelpers.saveTrip(newTrip);
+
+        return res.status(201).json({
+          status: 201,
+          message: 'Trip was created successfully.',
+          data: saveTrip
+        });
+      }
+      return res.status(401).json({
+        status: 401,
+        error: 'Only user with who have managers can make a trip request'
+      });
+    } catch (error) {
+      return res.status(500).json({
+        status: 500,
+        message: 'something goes wrong ',
+        error: error.message
+      });
+    }
+  }
+
+  /**
+   * This method handles view all trip requests.
+   * @param {object} req The user's request.
+   * @param {object} res The response.
+   * @param {function} next The next action.
+   * @returns {object} The status and some data of the trip.
+   */
+  static async viewAllTrips(req, res, next) {
+    try {
+      const { role, id } = req.user;
+      const {
+        start, end, pages, skip, paginate
+      } = await pagination.paginateData(req.query);
+      const foundTrips = await tripHelpers.findTripByRole(role, id, skip, start);
+      const userAllData = foundTrips.rows;
+      const countUserData = foundTrips.count;
+      if (foundTrips.rows.length === 0) {
+        return res.status(404).json({
+          status: 404,
+          message: `${req.user.email} You don't have trip request`
```

Comment:
No need to use the email. Brian is refactoring so that we can use **username** instead. Change the message to **You don’t currently have any trip requests**